synfig-studio/src/gui/dialogs/dialog_guide.cpp

```diff
@@ -0,0 +1,261 @@
+	guide_box->add(*guideGrid);
+//	guide_box->add(*coordFrame);
+//	guide_box->add(*coordGrid);
+	guide_box->set_margin_bottom(5);
+
+	//Box end
+	get_content_area()->pack_start(*guide_box);
+
+	Gtk::Button *apply_button(manage(new Gtk::Button(_("_Apply"), true)));
+	apply_button->show();
+	add_action_widget(*apply_button,0);
+	apply_button->signal_clicked().connect(sigc::bind(sigc::mem_fun(*this, &Dialog_Guide::on_ok_or_apply_pressed), false));
+
+	Gtk::Button *ok_button(manage(new Gtk::Button(_("_OK"), true)));
+	ok_button->show();
+	add_action_widget(*ok_button,1);
+	ok_button->signal_clicked().connect(sigc::bind(sigc::mem_fun(*this, &Dialog_Guide::on_ok_or_apply_pressed), true));
+
+	guide_box->show_all();
+
+//	current_work_area = work_area;
+}
+
+Dialog_Guide::~Dialog_Guide()
+{
+}
+
+void
+Dialog_Guide::set_current_guide_and_init(GuideList::iterator current_guide)
+{
+	curr_guide = current_guide;
+	init_widget_values();
+}
+
+
+void
+Dialog_Guide::on_ok_or_apply_pressed(bool ok)
+{
+	if (synfig::Angle::deg((*curr_guide).angle).get() != angle_widget->get_value() && degrees) {
+		(*curr_guide).angle = synfig::Angle::deg(angle_widget->get_value());
+	} else if ((*curr_guide).angle.get() != angle_widget->get_value() && !degrees) {
+		(*curr_guide).angle = synfig::Angle::rad(angle_widget->get_value());
+	} /*else
+		set_new_coordinates();*/
+
+	if (ok)
+		hide();
+	else
+		current_work_area->get_drawing_area()->queue_draw();
+}
+
+void
+Dialog_Guide::set_angle_type()
+{
+	Glib::ustring text = angle_type_picker.get_active_text();
+	if (text == "Degree")
+		degrees=true;
+	else if (text == "Radian")
+		degrees=false;
+}
+
+/*
+void
+Dialog_Guide::set_new_coordinates()
+{
+	float center_x_new = synfig::Distance(center_x_widget->get_value() , App::distance_system).get(synfig::Distance::SYSTEM_UNITS, canvas->rend_desc());
+
+	float center_y_new = synfig::Distance(center_y_widget->get_value() , App::distance_system).get(synfig::Distance::SYSTEM_UNITS, canvas->rend_desc());
+
+	float point_x_new = synfig::Distance(point_x_widget->get_value() , App::distance_system).get(synfig::Distance::SYSTEM_UNITS, canvas->rend_desc());
+
+	float point_y_new = synfig::Distance(point_y_widget->get_value() , App::distance_system).get(synfig::Distance::SYSTEM_UNITS, canvas->rend_desc());
+
+	if ((*curr_guide).isVertical) {
+		float slope = (point_y_new - (*curr_guide).point[1])/(point_x_new - center_x_new);
+		(*curr_guide).point[0] = center_x_new;
+		(*curr_guide).angle = synfig::Angle::rad(atan(slope));
+	} else {
+		float slope = (point_y_new - center_y_new)/(point_x_new - (*curr_guide).point[0]);
+		(*curr_guide).point[1] = center_y_new;
+		(*curr_guide).angle = synfig::Angle::rad(atan(slope));
+	}
+}
+*/
+
+void
+Dialog_Guide::init_widget_values()
+{
+//	float center_x = (*curr_guide).point[0];
+//	float center_y = (*curr_guide).point[1];
+//	float rotated_x = center_x + 2;
+//	float rotated_y = center_y + (2.0)*(tan((*curr_guide).angle.get()));
+
+	if(degrees)
```

Comment:
Just for safety, check if `curr_guide` is nullptr

Reply:
Actually I believe we dont really have to as `curr_guide` is not a pointer but rather an iterator. What do you think ?

Reply:
Oh. Although iterators can be invalid, there is no need here indeed :D